app/controllers/v0/sessions_controller.rb

```diff
@@ -136,22 +136,32 @@ def handle_completed_slo
       session         = Session.find(logout_request&.token)
       user            = User.find(session&.uuid)
 
+      # see comments for this method.
+      destroy_user_session!(user, session, logout_request)
       errors = build_logout_errors(logout_response, logout_request, session, user)
 
       if errors.size.positive?
         extra_context = { in_response_to: logout_response&.in_response_to }
         log_message_to_sentry("SAML Logout failed!\n  " + errors.join("\n  "), :error, extra_context)
         redirect_to Settings.saml.logout_relay + '?success=false'
       else
-        logout_request.destroy
-        session.destroy
-        user.destroy
         redirect_to Settings.saml.logout_relay + '?success=true'
-        # even if mhv logout raises exception, still consider logout successful from browser POV
-        MHVLoggingService.logout(user)
       end
     end
 
+    # FIXME: IMPORTANT - in the future, these should be destroyed even before the callback returns.
+    # if a user clicks on the link, we should destroy the session right away, who cares if callback returns?
+    # currently in production destroy fetches a link, but in the future it will be a redirect that triggers
+    # the flow right away. We need to ensure these are are always attempted regardless of the callback state!
```

Comment:
> if a user clicks on the link, we should destroy the session right away, who cares if callback returns?

The link takes the browser directly to ID.me, not to `api.vets.gov` which is why, in it's current form, `vets-api` must wait for the response.

> in the future it will be a redirect that triggers the flow right away

This could be problematic from a XSS perspective; ID.me might not like that Vets.gov would be the `referrer`.  Another solution could be to make the clicking of the logout button do two things:

1. Send a `DELETE` request to some new `vets-api` API
2. Follow the ID.me SLO link as it does today


Lastly, might I recommend filing a new issue for this rather than a multi-sentence comment in the code?

Reply:
So the plan for getting around analytics with vets.gov being the referrerer is to try passing a query parameter in the request that that the Backend does in the redirect. They should have a reference to the existing user either via GA analytics tracking id, or we might be able to modify headers accordingly.

This is a temporary bug fix. I have a separate PR where I am refactoring SLO into an SLO service similar to SSO service, but want to focus on redoing our specs in sessions_controller first.

Reply:
> I think we can destroy session and user when the link is fetched and not have to wait for the callback to return

This sounds good to me.  However, this makes an assumption that once the logout link is fetched, that logout is occurring.  [This issue in Sentry](http://sentry.vetsgov-internal/vets-gov/platform-api-production/issues/593/) leads me to believe that this is currently not the case in `vets-website`.  I _believe_ that the logout link is being fetched preemptively somewhere (either on login or page refresh), presumably to increase logout performance.  Destroying the session on link fetch will require `vets-website` to be updated as well.

All of this of course being a tangent to this PR :smiley:

Reply:
> I am happy to open as a separate ticket, but this is a temporary bug fix

![image](https://user-images.githubusercontent.com/3077884/37112457-7a583c02-2210-11e8-88bf-f569b30ba7ea.png)

Reply:
Tanel just weighed in on the analytics concern in #idme-vetsgov

> we have no visibility into anything going on your domain anyway
we can only see the flow from the point it hits our TLD until it leaves that
and we determine all of the information about where its coming from etc. based on the payload itself